File: quicksort/quicksort.go
package quicksort

func partition(numbers []int, low, high int) int {
	i := low
	pivot := numbers[high]
	for j := low; j < high; j++ {
		if numbers[j] < pivot {
			// Use go way to swap numbers instead of longer version in other languages.
			numbers[i], numbers[j] = numbers[j], numbers[i]
			i += 1
		}
	}
	numbers[i], numbers[high] = numbers[high], numbers[i]
	return i
}

func quicksort(numbers []int, low, high int) {
	if low < high {
		p := partition(numbers, low, high)
		quicksort(numbers, low, p-1)
		quicksort(numbers, p+1, high)
	}
}

File: stringAlgorithms/quicksort/quicksort.js
"use strict";

// temporary variable swap technique in JavaScript
function swap(numbers, i, j) {
    let temp = numbers[i];
    numbers[i] = numbers[j];
    numbers[j] = temp;
 }

function partition(numbers, low, high) {
    let i = low;
    let pivot = numbers[high];
    for (let j = low; j < high; j++) {
        if (numbers[j] < pivot) {
            swap(numbers, i, j);
            i++
        }
    }
    swap(numbers, i, high);
	return i
}

function quicksort(numbers, low, high) {
	if (low < high) {
		let p = partition(numbers, low, high)
		quicksort(numbers, low, p-1)
		quicksort(numbers, p+1, high)
    }
    return numbers;
}

exports.quicksort = quicksort;